phoenix-core/src/main/java/org/apache/phoenix/iterate/ExplainTable.java

```diff
@@ -204,15 +205,19 @@ private void appendPKColumnValue(StringBuilder buf, byte[] range, Boolean isNull
             range = ptr.get();
         }
         if (changeViewIndexId) {
-            Short s = (Short) type.toObject(range);
-            s = (short) (s + (-Short.MAX_VALUE));
```

Comment:
The offset is just for readability. Instead of displaying -32,768, we display 0. For shared indexes (i.e. indexes on views or local indexes), we do this. It's not necessary functionally.